### ISSUE TITLE
Installation error on python 3.11 macOS

### ISSUE THREAD

**Issue by pranftw** [onnx/onnx#4631]
# Bug Report

### Is the issue related to model conversion?

No

### Describe the bug

Unable to pip install onnx on macOS Monterey python 3.11

```
Collecting onnx
  Downloading onnx-1.12.0.tar.gz (10.1 MB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 10.1/10.1 MB 3.9 MB/s eta 0:00:00
  Preparing metadata (setup.py) ... error
  error: subprocess-exited-with-error
  
  × python setup.py egg_info did not run successfully.
  │ exit code: 1
  ╰─> [7 lines of output]
      fatal: not a git repository (or any of the parent directories): .git
      Traceback (most recent call last):
        File "<string>", line 2, in <module>
        File "<pip-setuptools-caller>", line 34, in <module>
        File "/private/var/folders/lf/r17346zd2gl5jn_70c1pjcr40000gn/T/pip-install-mg1yd6z8/onnx_2dc3bc8e91ae4f2695a13c0adb4afb54/setup.py", line 81, in <module>
          assert CMAKE, "Could not find cmake executable!"
      AssertionError: Could not find cmake executable!
      [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
error: metadata-generation-failed

× Encountered error while generating package metadata.
╰─> See above for output.

note: This is an issue with the package mentioned above, not pip.
hint: See above for details.
```

### System information
- OS Platform and Distribution (*e.g. Linux Ubuntu 16.04*): macOS Monterey 12.6.1
- ONNX version (*e.g. 1.7*):  1.12.0
- Python version: 3.11
- GCC/Compiler version (if compiling from source):
  Apple clang version 14.0.0 (clang-1400.0.29.102)
  Target: arm64-apple-darwin21.6.0
  Thread model: posix
  InstalledDir: /Library/Developer/CommandLineTools/usr/bin

### Reproduction instructions
`pip install onnx`

### Expected behavior
Install successfully

**Comment by daquexian** [comment]
onnx 1.12 doesn't have prebuilt wheel for Python 3.11. Maybe we can look forward to the support in 1.13 https://github.com/onnx/onnx/pull/4490

And you might want to install cmake by `pip install cmake` to bypass the current problem.

**Comment by jcwchen** [comment]
Hi @pranftw,
As @daquexian said, currently ONNX doesn't have prebuilt wheel for Python 3.11 so actually you were building ONNX from source and cmake is required. Still, probably current ONNX cannot work with Python 3.11, because the submodule of pybind11 needs to be updated. (See https://github.com/onnx/onnx/pull/4490#issuecomment-1272257487)

**Comment by pranftw** [comment]
Thanks @daquexian @jcwchen Looking forward to when the next version will come out

**Comment by jcwchen** [comment]
Fresh ONNX 1.13 has been just released with Python 3.11 support: https://pypi.org/project/onnx/1.13.0/#files. I will close this issue now. Please let me know if you see any issues. Thanks!
